src/Protocol/Protocol_1_12.cpp

```diff
@@ -1019,10 +1019,23 @@ UInt32 cProtocol_1_12::GetPacketID(cProtocol::ePacketType a_Packet)
 
 
 
+void cProtocol_1_12::HandleCraftRecipe(cByteBuffer & a_ByteBuffer)
+{
+	HANDLE_READ(a_ByteBuffer, ReadBEUInt8, UInt8, WindowID);
+	HANDLE_READ(a_ByteBuffer, ReadVarInt, UInt32, RecipeID);
+	HANDLE_READ(a_ByteBuffer, ReadBool,          bool,    MakeAll);
+	m_Client->HandleCraftRecipe(RecipeID);
```

Comment:
No protocol specific IDs should escape `cProtcol` classes. You need to translate protocol id -> cuberite id here.